docs/content/reference/resource-schema/core-schema/gateway/snippets/gateway.bicep

```
@@ -32,10 +32,31 @@ resource gateway 'Applications.Core/gateways@2022-03-15-privatepreview' = {
         destination: backendroute.id
       }
     ]
+    tls: {
+      // Specify SSL Passthrough for your app (default: false)
+      sslPassthrough: false
+
+      // Specify TLS Termination for your app. Mutually exclusive with SSL Passthrough.
+      // Hostname for TLS termination
+      hostname: 'hostname.radapp.dev'
+      // The Radius Secret Store holding TLS certificate data
+      certificateFrom: secretstore.id
+      // The minimum TLS protocol version to support. Defaults to 1.2
+      minimumProtocolVersion: '1.2'
+    }
   }
 }
 //GATEWAY
 
+resource secretstore 'Applications.Core/secretStores@2022-03-15-privatepreview' = {
+  name: 'secretstore'
+  properties: {
+    application: app.id
+    data: {
+    }
+  }
```

@willdavsmith  @willtsai  I assume that gateway checks the type of secret and properties. Didn't we validate it ? 

```suggestion
  type: 'certificate'
  properties: {
    application: app.id
    data: {
        'tls.crt': {}
        'tls.key': {}
    }
    resource: '<secret key name>'
  }
```

---

Should this type be above properties or within it?

---

Closing as this isn't displayed to the user, and is only included so `secretStore.id` is available